### PR TITLE
docs: create remember-viewing-state usecase

### PR DESCRIPTION
**Pull request by tschawax**
References #1

## Basics

- [x] I added a line to [changelog.md](/doc/changelog.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages and changelog.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added automated tests or a [manual test protocol](../doc/tests/manual/protocol.md)
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)
- [ ] Exceptions to any guidelines are documented

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] I've read through the whole documentation
- [ ] I've checked conformity to guidelines

### CODE REVIEW

File: doc/usecases/draft/remember_viewing_state.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    - selected layer
+    - all layers' visibility
+    - grid display
+    - plant labels display
+- **Non-functional Constraints:**
+  - Changes to maps while being in offline-mode, are not considered for restoring (for now).
+  - _Zustand_'s [_persist middleware_](https://docs.pmnd.rs/zustand/integrations/persisting-store-data) should be used to store a map's state in a storage of any kind - not only do we already use _Zustand_ as our global state management library, but it also supports de-/serialization of `Maps` and `Sets`.  
+    _Persist_'s [_partialize_](https://docs.pmnd.rs/zustand/integrations/persisting-store-data#partialize) option should then be used to only store the 'restore-worthy' data of a map.
+  - Copy & Paste related data should not be restored, to not only reduce complexity of the storing/restoring- as well as the undo/redo-process, but also to prevent out-of-sync scenarios where copied elements are already deprecated/removed/outdated in the system/database.
+    Besides that, applications offering the recovery of Copy & Paste state in a new session are extremely rare and therefore something the user does not necessarily expect to get.
```
Comment (markus2330) [inline doc/usecases/draft/remember_viewing_state.md:55]: :+1:

**Review by markus2330 — CHANGES_REQUESTED** [review]
LGTM, only minor questions.

File: doc/usecases/draft/remember_viewing_state.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    - map's (x,y)-offset in the viewport
+    - selected layer
+    - all layers' visibility
+    - grid display
+    - plant labels display
+- **Non-functional Constraints:**
+  - Changes to maps while being in offline-mode, are not considered for restoring (for now).
+  - _Zustand_'s [_persist middleware_](https://docs.pmnd.rs/zustand/integrations/persisting-store-data) should be used to store a map's state in a storage of any kind - not only do we already use _Zustand_ as our global state management library, but it also supports de-/serialization of `Maps` and `Sets`.  
+    _Persist_'s [_partialize_](https://docs.pmnd.rs/zustand/integrations/persisting-store-data#partialize) option should then be used to only store the 'restore-worthy' data of a map.
+  - Copy & Paste related data should not be restored, to not only reduce complexity of the storing/restoring- as well as the undo/redo-process, but also to prevent out-of-sync scenarios where copied elements are already deprecated/removed/outdated in the system/database.
```
Comment (badnames) [inline doc/usecases/draft/remember_viewing_state.md:54]: These questions are not directly related to this usecase, but just so I'm reading this right:
Does that mean we are not using the OS's system clipboard on the map?
If we are, can the user copy/paste between different browser tabs or different maps in the same tab?

Comment (markus2330) [reply doc/usecases/draft/remember_viewing_state.md:54]: Interesting remarks, I agree it should be clarified what will work when done that way.

I just discussed it with @yvonnemarkl and a few remarks from our side: It is okay that the OS clipboard is not used and that copying between tabs does not work if then the implementation is easier that way. (Importing/exporting is better done in a different way and people most probably don't expect that they can paste into other apps.)

File: doc/usecases/draft/remember_viewing_state.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  The user is logged in and has opened a map.
+- **Main success scenario:**
+  - The user selects layer _A_.  
+    He zooms into the map.  
+    He scrolls/drags the map's viewport.  
+    He sets layer _B_ invisible.  
+    He turns off the grid display.  
+    He hides the plant labels.  
+    He closes the browser, opens it again and logs in.
+    He opens the same map again which he edited in his previous session.
```
Comment (markus2330) [inline doc/usecases/draft/remember_viewing_state.md:23]:
```suggestion
  - He opens the same map again which he edited in his previous session.  
```

File: doc/usecases/draft/remember_viewing_state.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+- **Main success scenario:**
+  - The user selects layer _A_.  
+    He zooms into the map.  
+    He scrolls/drags the map's viewport.  
+    He sets layer _B_ invisible.  
+    He turns off the grid display.  
+    He hides the plant labels.  
+    He closes the browser, opens it again and logs in.
+    He opens the same map again which he edited in his previous session.  
+    He finds all the changes and settings from the last session being restored:
```
Comment (markus2330) [inline doc/usecases/draft/remember_viewing_state.md:24]:
```suggestion
  - He finds all the changes and settings from the last session being restored:
```

**Review by markus2330 — APPROVED** [review]
Great job, very detailed :rocket:

File: doc/usecases/draft/remember_viewing_state.md (hunk 3 above)
Comment (markus2330) [reply doc/usecases/draft/remember_viewing_state.md:23]: Ok, if you don't like sub-itemize then let us leave it with line breaks.
